dataclients/kubernetes/kube.go

```diff
@@ -481,15 +483,46 @@ func (c *Client) ingressToRoutes(items []*ingressItem) ([]*eskip.Route, error) {
 					}
 
 					r.HostRegexps = host
-					routes = append(routes, r)
+					hostRoutes = append(hostRoutes, r)
 				}
 			}
+
+			routes = append(routes, hostRoutes...)
+
+			// if routes were configured, but there is no catchall route defined,
+			// create a route which returns 404
+			if len(hostRoutes) > 0 && !catchAllRoutes(hostRoutes) {
+				catchAll := &eskip.Route{
+					Id:          routeID(i.Metadata.Namespace, i.Metadata.Name, rule.Host, "", ""),
+					HostRegexps: host,
+					BackendType: eskip.ShuntBackend,
```

Comment:
@aryszka you mentioned in #408 to use the status filter. Is that better than using the `ShuntBackend` like I do here?

Reply:
@mikkeloscar the shunt backend is fine, there must be a backend, and shunt is like a "blind" backend. The status filter can be used in addition to that. But if the default status is good, then the status filter is not needed. So, I think it is good like this.

Reply:
>  But if the default status is good, then the status filter is not needed.

Since this does not use status filter it would use DefaultHTTPStatus - the same value as when route lookup failed which means that catchall routes are useless (unless I am missing something) - the client would get 404 (or configured value) either way.

Reply:
@AlexanderYastrebov the problem is something like "confused deputy". So you create a prepared call to x, but get application y because of no catchall host route. I think this was the reasoning why we have a catchall by host. IIRC one driver was, because of the ingress route annotation and predicate annotation.
Since RouteGroup I think we have everything to replace all these use cases, so I would also say we can drop the catchall for host routes.